nixos/roles/prometheus/default.nix

```diff
@@ -0,0 +1,139 @@
+                };
+                enable = lib.mkEnableOption (lib.mdDoc "Enable alert rule") // {
+                  default = true;
+                };
+                expr = lib.mkOption {
+                  type = lib.types.str;
+                };
+                for = lib.mkOption {
+                  type = lib.types.str;
+                  default = "2m";
+                };
+                labels = lib.mkOption {
+                  type = lib.types.attrsOf lib.types.str;
+                  default = { };
+                };
+                annotations = lib.mkOption {
+                  type = lib.types.attrsOf lib.types.str;
+                  default = { };
+                };
+              };
+            }));
+            default = { };
+          };
+          recordingRules = lib.mkOption {
+            type = lib.types.attrsOf (lib.types.submodule ({ name, ... }: {
+              options = {
+                name = lib.mkOption {
+                  type = lib.types.str;
+                  default = name;
+                };
+                enable = lib.mkEnableOption (lib.mdDoc "Enable recording rule") // {
+                  default = true;
+                };
+                expr = lib.mkOption {
+                  type = lib.types.str;
+                };
+                for = lib.mkOption {
+                  type = lib.types.str;
+                  default = "2m";
+                };
+                labels = lib.mkOption {
+                  type = lib.types.attrsOf lib.types.str;
+                  default = { };
+                };
+                annotations = lib.mkOption {
+                  type = lib.types.attrsOf lib.types.str;
+                  default = { };
+                };
+              };
+            }));
+            default = { };
+          };
+        };
+      }));
+      example = {
+        prometheusAlerts = {
+          alertRules = {
+            ExampleAlert = {
+              expr = "up == 0";
+              for = "4m";
+              labels = {
+                severity = "critical";
+              };
+              annotations = {
+                summary = "Instance {{ $labels.instance }} down";
+                description = "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 2 minutes.";
+              };
+            };
+          };
+          recordingRules = {
+            RabbitmqDeliveredMessages = {
+              expr = "rate(rabbitmq_queue_messages_delivered_total[5m])";
+              annotations = {
+                description = "The rate of messages delivered to queues over the last 5 minutes";
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+  config = {
+    services.prometheus = {
+      enable = lib.mkDefault true;
+      # checks fail because of missing secrets in the sandbox
+      checkConfig = false;
+      ruleFiles = [ (pkgs.writers.writeYAML "rules.yaml" rules) ];
```

Comment:
Following the documentation, you may set it to `"syntax-only"` to still have some checking.

> If you use credentials stored in external files (password_file, bearer_token_file, etc), they will not be visible to promtool and it will report errors, despite a correct configuration. To resolve this, you may set this option to "syntax-only" in order to only syntax check the Prometheus configuration.
> https://github.com/NixOS/nixpkgs/blob/475d5ae2c4cb87b904545bdb547af05681198fcc/nixos/modules/services/monitoring/prometheus/default.nix#L1738-L1743

Reply:
applied.